src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -221,12 +221,21 @@ public void InitializeFullRange(DateTime start, DateTime end)
             _init = true;
         }
 
-        private void InitByTicks(DateTime start)
+        private void InitByTicks(DateTime first)
         {
-            var ticks = start.Ticks;
+            var ticks = first.Ticks;
             ticks -= (ticks % AlignedTicks);
-            _start = new DateTime(ticks, start.Kind);
+
+            if (ticks > _start.Ticks)
+            {
+                double distance = ticks - _start.Ticks;
+                var ticksToAdd = (long)Math.Floor(distance / Ticks) * Ticks;
```

Comment:
This should yield same result without type conversions, right?
`var distance = ticks - _start.Ticks; ` // keep it long
`var ticksToAdd = ((distance / Ticks) * Ticks)`